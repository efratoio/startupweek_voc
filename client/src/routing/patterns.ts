import * as UrlPattern from "url-pattern";

export const patternLogin = new UrlPattern("/login(/)");
export const patternDashboard = new UrlPattern("/dashboard(/)");
export const patternSignup = new UrlPattern("/signup(/)");
