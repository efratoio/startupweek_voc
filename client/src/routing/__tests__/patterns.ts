import * as patterns from "../patterns";

[
    {
        pattern: "patternLogin",
        okay: ["/login", "/login/"],
        fail: ["/signup", "/login?test=example", "", "/loginexample"],
    },
    {
        pattern: "patternSignup",
        okay: ["/signup", "/signup/"],
        fail: ["/login", "/signup?test=example", "", "/signupexample"],
    },
    {
        pattern: "patternDashboard",
        okay: ["/dashboard", "/dashboard/"],
        fail: ["/signup", "/dashboard?test=example", "", "/dashboardexample"],
    },
].forEach(({ pattern, okay, fail }) => {
    okay.forEach(url => {
        test(`The URL pattern "${pattern}" with the URL "${url}" returns the expected result`, () => {
            expect(patterns[pattern].match(url)).toMatchSnapshot();
        });
    });
    fail.forEach(url => {
        test(`The URL pattern "${pattern}" with the URL "${url}" returns \`undefined\``, () => {
            expect(patterns[pattern].match(url)).toBeNull();
        });
    });
});
