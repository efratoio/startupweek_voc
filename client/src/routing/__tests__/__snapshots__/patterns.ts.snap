// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The URL pattern "patternDashboard" with the URL "/dashboard" returns the expected result 1`] = `Object {}`;

exports[`The URL pattern "patternDashboard" with the URL "/dashboard/" returns the expected result 1`] = `Object {}`;

exports[`The URL pattern "patternLogin" with the URL "/login" returns the expected result 1`] = `Object {}`;

exports[`The URL pattern "patternLogin" with the URL "/login/" returns the expected result 1`] = `Object {}`;

exports[`The URL pattern "patternSignup" with the URL "/signup" returns the expected result 1`] = `Object {}`;

exports[`The URL pattern "patternSignup" with the URL "/signup/" returns the expected result 1`] = `Object {}`;
